docs/Spring Boot/5_Step_To_Make_Gradle_Configuration_Extreme_Clean_In_A_Multi-Module_Project.md

```diff
@@ -0,0 +1,216 @@
+
+1. extract version declaring in `gradle.properties`.
+2. define all plugin and repository in `settings.gradle`.
+3. define all library in the `allprojects.dependencyManagement` in `./build.gradle`.
+4. declaring dependency and plugin directly instead of using `subproject` in submodule.
+5. extract complex and common task config to extra file and apply wherever need it.
+
+Take a look at this [repository](https://github.com/NoahHsu/event-sourcing-order-poc) or [refactor PR](https://github.com/NoahHsu/event-sourcing-order-poc/pull/72), if you can't wait to find out how it looks.
+
+## Step by Step Demonstration
+
+### Step 1: Extract Version Declaration
+
+Version declarations can be extracted into a `gradle.properties` file. Additionally, Gradle arguments can be defined as shown below:
+
+```properties title="./gradle.properties"
+group='org.example'
+version=0.0.1.SNAPSHOT
+
+# Plugin Version
+jibVersion=3.4.3
+
+# Spring Version
+springBootVersion=3.1.5
+springDependencyVersion=1.1.4
+springCloudVersion=2022.0.1
+
+# Dependency Version
+springdocVersion=2.1.0
+feignMicrometerVersion=12.1
+wiremockVersion=3.7.0
+logbackAppenderVersion=1.4.0-rc2
+lombokVersion=1.18.20
+
+# Gradle Argument
+org.gradle.parallel=true
+```
+
+### Step 2: Define Used Plugins and Maven Source 
+
+All used plugins and the source Maven repository can be defined in a `settings.gradle`.:
+
+```groovy title="./settings.gradle"
+import org.gradle.api.initialization.resolve.RepositoriesMode
+
+pluginManagement {
+    plugins {
+        id 'org.springframework.boot' version "${springBootVersion}"
+        id 'io.spring.dependency-management' version "${springDependencyVersion}"
+        id 'com.google.cloud.tools.jib' version "${jibVersion}"
+    }
+}
+
+dependencyResolutionManagement {
+    repositoriesMode.set(RepositoriesMode.FAIL_ON_PROJECT_REPOS)
+    repositories {
+        mavenCentral()
+        mavenLocal()
+        maven {
+            url '.m2/local/'
+        }
+    }
+}
+
+rootProject.name = 'event-sourcing-order-poc'
+
+include 'modules'
+include 'modules:common'
+findProject(':modules:common')?.name = 'common'
+// ... and other modules settings
+
+include 'order'
+include 'order:command-side'
+findProject(':order:command-side')?.name = 'order-command-side'
+include 'order:event-handler'
+findProject(':order:event-handler')?.name = 'order-event-handler'
+include 'order:query-side'
+findProject(':order:query-side')?.name = 'order-query-side'
+// ... and other sub-project settings
+
+```
+
+### Step 3: Define Allprojects DependencyManagement
+
+All the used libraries should be defined in a `allprojects.dependencyManagement` closure in `build.gradle` of the root module:
```

Comment:
📝 **[vale]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[write-good.E-Prime] Try to avoid using 'be'.